src/test/scala/de/frosner/ddq/core/CheckTest.scala

```diff
@@ -489,10 +489,25 @@ class CheckTest extends FlatSpec with Matchers with BeforeAndAfterEach with Befo
     val ref = makeIntegersDf(List(1, 3, 100), List(1, 500, 100))
     val check = Check(base).isJoinableWith(ref, "column1" -> "column1", "column3" -> "column2")
     val constraint = check.constraints.head
-    val result = ConstraintSuccess("Columns column1->column1, column3->column2 can be used for joining (one distinct row match)")
+    val result = ConstraintSuccess("Columns column1->column1, column3->column2 can be used for joining (number of distinct rows in base table: 2, number of distinct rows after joining: 1, unmatched keys in base table: 50%)")
     check.run().constraintResults shouldBe Map(constraint -> result)
   }
 
+  it should "compute the unmatched keys in base table correctly" in {
+    val base = makeIntegerDf(List(1, 1, 1, 1, 1, 1, 1, 1, 1, 2))
+    val ref = makeIntegerDf(List(1))
+
+    val check1 = Check(base).isJoinableWith(ref, "column" -> "column")
+    val constraint1 = check1.constraints.head
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/frank_7/drunken-data-quality/file/1994540670/issues/source?bid=3033737&fileBranchId=3048638#l501)